scripts/machine-launch-script.sh

```diff
@@ -0,0 +1,305 @@
+#!/bin/bash
+
+CONDA_INSTALL_PREFIX=/opt/conda
+CONDA_INSTALLER_VERSION=4.12.0-0
+CONDA_INSTALLER="https://github.com/conda-forge/miniforge/releases/download/${CONDA_INSTALLER_VERSION}/Miniforge3-${CONDA_INSTALLER_VERSION}-Linux-x86_64.sh"
+CONDA_CMD="conda" # some installers install mamba or micromamba
+CONDA_ENV_NAME="chipyard"
+
+DRY_RUN_OPTION=""
+DRY_RUN_ECHO=()
+REINSTALL_CONDA=0
+
+usage()
+{
+    echo "Usage: $0 [options]"
+    echo
+    echo "Options:"
+    echo "[--help]                  List this help"
+    echo "[--prefix <prefix>]       Install prefix for conda. Defaults to /opt/conda."
+    echo "                          If <prefix>/bin/conda already exists, it will be used and install is skipped."
+    echo "[--env <name>]            Name of environment to create for conda. Defaults to '$CONDA_ENV_NAME'."
+    echo "[--dry-run]               Pass-through to all conda commands and only print other commands."
+    echo "                          NOTE: --dry-run will still install conda to --prefix"
+    echo "[--reinstall-conda]       Repairs a broken base environment by reinstalling."
+    echo "                          NOTE: will only reinstall conda and exit without modifying the --env"
+    echo
+    echo "Examples:"
+    echo "  % $0"
+    echo "     Install into default system-wide prefix (using sudo if needed) and add install to system-wide /etc/profile.d"
+    echo "  % $0 --prefix ~/conda --env my_custom_env"
+    echo "     Install into $HOME/conda and add install to ~/.bashrc"
+    echo "  % $0 --prefix \${CONDA_EXE%/bin/conda} --env my_custom_env"
+    echo "     Create my_custom_env in existing conda install"
+    echo "     NOTES:"
+    echo "       * CONDA_EXE is set in your environment when you activate a conda env"
+    echo "       * my_custom_env will not be activated by default at login see /etc/profile.d/conda.sh & ~/.bashrc"
```

Comment:
I don't like that this script is needs sudo to do all these system-level changes. Basically the types of users are:
- Users running on a disposable F1 instance
- Users running on a managed cluster
- Users running on a personal computer

Only for the F1 use case is changing system-level settings acceptable.

Reply:
It only uses sudo and changes the system-level settings if you tell it to install at a path that can't be written by the current user or you run the script as root.  That should be clarified.